Scripts/Core/EditorVR.Menus.cs

```diff
@@ -197,119 +109,282 @@ internal void UpdateMenuVisibilities()
+								var otherMenus = otherHideFlags.Keys.ToList();
+								foreach (var menu in otherMenus)
+								{
+									// Only set if hidden--value is reset every frame
+									otherHideFlags[menu] |= MenuHideFlags.HasDirectSelection;
+								}
+								break;
+							}
+						}
+					}
+				}
+
+				// Set show/hide timings
+				foreach (var deviceData in m_ActiveDeviceData)
+				{
+					var menuHideFlags = deviceData.menuHideFlags;
+					var menus = menuHideFlags.Keys.ToList();
+					foreach (var menu in menus)
+					{
+						var hideFlags = menuHideFlags[menu];
+						if ((hideFlags & ~MenuHideFlags.Hidden & ~MenuHideFlags.OtherMenu) == 0)
+							deviceData.menuAutoHideTimes[menu] = Time.time;
+
+						if (hideFlags != 0)
+							deviceData.menuAutoShowTimes[menu] = Time.time;
 					}
 				}
 
 				// Apply state to UI visibility
-				Rays.ForEachProxyDevice(deviceData =>
+				foreach (var deviceData in m_ActiveDeviceData)
 				{
 					var mainMenu = deviceData.mainMenu;
-					mainMenu.visible = deviceData.menuHideFlags[mainMenu] == 0;
+					var mainMenuHideFlags = deviceData.menuHideFlags[mainMenu];
+					if (mainMenuHideFlags != 0 && !mainMenu.hovering)
+					{
+						if ((mainMenuHideFlags & MenuHideFlags.Hidden) != 0)
+						{
+							mainMenu.visible = false;
+						}
+						else if (Time.time > deviceData.menuAutoHideTimes[mainMenu] + k_MainMenuAutoHideDelay)
+						{
+							mainMenu.visible = false;
+						}
+					}
+					else if (Time.time > deviceData.menuAutoShowTimes[mainMenu] + k_MainMenuAutoShowDelay)
+					{
+						mainMenu.visible = true;
+					}
+
+					deviceData.mainMenuActivator.disabled = (mainMenuHideFlags & MenuHideFlags.Temporary) != 0;
 
 					var customMenu = deviceData.customMenu;
 					if (customMenu != null)
 						customMenu.visible = deviceData.menuHideFlags[customMenu] == 0;
 
 					UpdateAlternateMenuForDevice(deviceData);
 					Rays.UpdateRayForDevice(deviceData, deviceData.rayOrigin);
-				});
-
-				evr.GetModule<DeviceInputModule>().UpdatePlayerHandleMaps();
-			}
+				}
 
-			internal static void OnMainMenuActivatorHoverStarted(Transform rayOrigin)
-			{
-				var deviceData = evr.m_DeviceData.FirstOrDefault(dd => dd.rayOrigin == rayOrigin);
-				if (deviceData != null)
+				// Reset Temporary states
+				foreach (var deviceData in m_ActiveDeviceData)
 				{
-					var menus = new List<IMenu>(deviceData.menuHideFlags.Keys);
+					var menuHideFlags = deviceData.menuHideFlags;
+					var menus = menuHideFlags.Keys.ToList();
 					foreach (var menu in menus)
 					{
-						deviceData.menuHideFlags[menu] |= MenuHideFlags.OverActivator;
+						menuHideFlags[menu] &= ~MenuHideFlags.Temporary;
 					}
 				}
+
+				evr.GetModule<DeviceInputModule>().UpdatePlayerHandleMaps();
 			}
 
-			internal static void OnMainMenuActivatorHoverEnded(Transform rayOrigin)
+			internal static bool IsValidHover(MultipleRayInputModule.RaycastSource source)
 			{
+				var go = source.draggedObject;
+				if (!go)
```

Comment:
This fix has been pushed, but I did !go on the line below, and it didn't mark the change.